### ISSUE TITLE
fromBeginning option only delivers content upon first change

### ISSUE THREAD

**Issue by nathankellenicki** [lucagrulla/node-tail#40]
Hi, thank you for the recent bug fix for fromBeginning. I've noticed that the behaviour only delivers the content when the first change was noticed.

Expected behaviour (At least, from my perspective) would be that it would deliver the existing contents of the file as soon as the stream is opened. Is this correct?

Cheers,
Nathan

**Comment by lucagrulla** [comment]
You are right, the current implementation needs one change to the file to deliver the "historical" content.

Given that at this stage the specific timing of that first delivery is undefined I would say that there's no actual expected behaviour.

**Comment by nathankunicki** [comment]
In which case, I'd say one should be defined either way.

For our particular use case we've written a tool to read stdout logs from a separate process to determine the current state in the process lifetime. On startup, the tool should read the current contents of the file to get the current state, and follow the log so that it can be updated with further developments. I'd argue that functionality to this is similar to what most people would expect, though without further input that's just conjecture. :)

If that's not the desired functionality, perhaps a option should be added to the constructor, "immediate: true" or the like, to determine whether it should return the existing contents immediately? Without this, one must find an alternative means to read in the current content.

In the meantime, I've made it trigger immediately like this (Somewhat hacky, but it works):

```
var tail = new Tail("log_filename", {
    fromBeginning: true
});

tail.on("line", function (line) {
    ...
});

tail.watchEvent.call(tail, "change");
```

**Comment by lucagrulla** [comment]
Most of people will observe file updates pretty much straight after tailing the file so they will be fine.
I'll flag this as a feature request but I'm not expecting to go work on this anytime soon.

As you observed as well I can't see a clean way to eventually add it as a feature, only hacky ones.
PRs are welcome, but they have to be pretty neat solutions :-)

cheers
L.

**Comment by tiagocury** [comment]
+1

**Comment by rohit-lakhanpal** [comment]
+1

**Comment by CiscoKidxx** [comment]
+1

**Comment by colbygk** [comment]
+1. Starting delivery immediately would be the path of least surprise.

**Comment by hotantuna** [comment]
+1

**Comment by geogeim** [comment]
+1 i'm using @nathankunicki 's workaround for now

**Comment by vitaly-nevgen** [comment]
Created PR with potential fix of this issue @lucagrulla @nathankunicki

**Comment by suculent** [comment]
Can you merge pls?

**Comment by mojo2405** [comment]
THuge thanks to @nathankunicki

**Comment by peterg19** [comment]
+1  Was definitely my initial expectation.  Thanks for the workaround.

**Comment by psi-4ward** [comment]
any progress?

**Comment by lucagrulla** [comment]
Not yet.

**Comment by lucagrulla** [comment]
Fixed with version 2.0.0
